pegasus/sites.v3/code.org/public/private/edit-csp-properties/splat.haml

```diff
@@ -0,0 +1,124 @@
+
+:javascript
+  'use strict';
+
+  // Declare app level module which depends on filters, and services
+  angular.module('propertiesApp', [
+    'ngRoute',
+    'ngResource',
+    'propertiesApp.controllers',
+    'propertiesApp.services'
+  ]).config(['$routeProvider', function($routeProvider) {
+    $routeProvider.when('/',
+                        {templateUrl: '/private/edit-csp-properties/properties',
+                         controller: 'PropertiesController'});
+    $routeProvider.otherwise({redirectTo: '/'});
+  }]);
+
+  // SERVICES
+  var services = angular.module('propertiesApp.services', []).
+      value('version', '0.1');
+
+  services.factory('propertiesService', ['$resource',
+    function($resource) {
+      var Property = $resource("#{api_base_url}/:key", {}, {
+        // default methods: see https://code.angularjs.org/1.2.21/docs/api/ngResource/service/$resource
+        //  'get':    {method: 'GET'},
+        //  'save':   {method: 'POST'},
+        //  'query':  {method: 'GET', isArray:true},
+        //  'remove': {method: 'DELETE'},
+        //  'delete': {method: 'DELETE'} // don't use this because it doesn't work in IE9
+      });
+
+      return Property;
+    }]);
+
+  // CONTROLLERS
+
+  var controllers = angular.module('propertiesApp.controllers', []).
+      value('version', '0.1');
+
+  controllers.controller('PropertiesController', ['$scope', '$http', '$route', '$routeParams', '$location', '$window', 'propertiesService',
+      function($scope, $http, $route, $routeParams, $location, $window, propertiesService) {
+    $scope.properties = propertiesService.get();
+    $scope.keys = null;
+    $scope.editing = [];
+    $scope.editingNewProperty = false;
+    $scope.newKey = '';
+    $scope.newValue = '';
+
+    $scope.properties.$promise.then(function(properties) {
+      console.log(properties);
+      $scope.keys = getKeys(properties);
+      console.log($scope.keys);
+    }).catch($scope.genericError);
+
+    $scope.genericError = function(result) {
+      $window.alert("An unexpected error occurred, please try again. If this keeps happening, try reloading the page.");
+    };
+
+    $scope.saveProperty = function(key) {
+      var value = castValue($scope.properties[key]);
+      $scope.properties[key] = value;
+      // The properties API doesn't quite fit Angular's $resource model.
+      $http.post("#{api_base_url}/" + key, JSON.stringify(value));
+      $scope.editing[key] = false;
+    }
+
+    $scope.deleteProperty = function(key) {
+      $scope.properties.$delete({key:key});
+      var index = $scope.keys.indexOf(key);
+      if (index > -1) {
+        $scope.keys.splice(index, 1);
+      }
+      $scope.editing[key] = false;
+    }
+
+    $scope.saveNewProperty = function() {
+      var key = $scope.newKey;
+      var value = castValue($scope.newValue);
+      $scope.properties[key] = value;
+      $scope.keys.push(key);
+      // The properties API doesn't quite fit Angular's $resource model.
```

Comment:
I couldn't figure out how to do it with $resource, because $save seems to always want to write an object rather than a literal value.